lib/pos_blockchain.go

```diff
@@ -134,8 +135,48 @@ func (bc *Blockchain) validateDeSoBlockPoS(desoBlock *MsgDeSoBlock) error {
 // that the timestamp is valid, that the version of the header is valid,
 // and other general integrity checks (such as not malformed).
 func (bc *Blockchain) validateBlockGeneral(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	// First make sure we have a non-nil header
+	if desoBlock.Header == nil {
+		return RuleErrorNilBlockHeader
+	}
+
+	// Timestamp validation
+	// What's the criteria? Is it just that it's not in the future?
+	// And that it's not too far in the past? Too far being older than the current best chain tip?
+	if desoBlock.Header.TstampNanoSecs < bc.GetBestChainTip().Header.TstampNanoSecs {
+		return RuleErrorPoSBlockTstampNanoSecsTooOld
+	}
+	if desoBlock.Header.TstampNanoSecs > uint64(time.Now().UnixNano()) {
```

Comment:
The current block can have a timestamp that's up to 10 minutes beyond the node's current timestamp. The "10 minute" threshold may be either configured DeSo param or an on-chain GlobalParam. Up to you which one to go with. Whatever is simpler seems fine.

```suggestion
	if desoBlock.Header.TstampNanoSecs > uint64(time.Now().UnixNano().Add(10 * time.Minute)) {
```

Reply:
I'll add it to the deso params and leave a TODO that we may want to move to global params for now.